expr/expr.go

```diff
@@ -146,6 +156,16 @@ func (e expr) consumeBasicArg(pos int, exp Arg) (int, error) {
 			return 0, ErrBadArgumentStr{"string", string(got.etype)}
 		}
 		*v.val = got.bool
+	case ArgStringsOrInts:
+		// special case! consume all subsequent args (if any) in args that will also yield a string
```

Comment:
why do ArgStrings and ArgStringsOrInts look different from the other multiple-arg cases? eg they don't validate the current element. do they allow there to be 0 args?

Reply:
Graphite allows aliasByNodes with no trailing args (sets target to "").  I didn't particular intend for this case. I'm open to making sure there is at least one valid argument following.

Reply:
hmm the graphite docs for both aliasByNode and aliasByTags both state "one or more"
http://graphite.readthedocs.io/en/latest/functions.html#graphite.render.functions.aliasByNode
let's check in with @DanCech to see if either graphite's docs should say "zero or more", or graphite implementation should change to require at least one.

Reply:
yeah we likely just need to add a validation check in graphite. that said, anyone calling alias functions without any specifications deserves what they get.

Reply:
@DanCech so to be clear, you're saying we should stick with "one or more" and can validate as such, as the fact that zero or more is currently allowed (for both aliasByNode and aliasByTags) is not something we actually want / should rely on / should implement in MT ?

Reply:
Right now it should follow graphite.  If we want to change the graphite behavior we can talk about that (PRs appreciated!)

Reply:
The MT parser doesn't allow 0 arguments for this parameter. I think this is fine, but it doesn't match graphite. I'd rather not muck with the parser to allow 0 args if we don't think that's the right thing to do.

Reply:
Ok, so this function won't even get called if there are 0 arguments. `ArgStrings` and `ArgStringsOrInts` can both be 0-length if the wrong type is encountered (e.g. `groupByTags(...,"sum",1,"tag2")` would stop at `1` and not encounter `"tag2"`. I think maybe changing the assumption to be that *all* subsequent args *must* be the right type or throw the error. That seems like the correct behavior.

Reply:
Interestingly, this would go against what `ArgInts` does today, but nothing seems to use that.